ggml/examples/unity/unity.cpp

```diff
@@ -179,46 +150,47 @@ int main(int argc, char ** argv) {
             if (interactive) continue;
             else return 1;
         }
-        auto tgt_lang_ptr = model.vocab.token_to_id.find("__" + tgt_lang + "__");
-        if (tgt_lang_ptr == model.vocab.token_to_id.end()) {
-            std::cerr << "Unknown language " << tgt_lang << "\n";
-            if (interactive) continue;
-            else return 2;
-        }
-        int tgt_lang_idx = tgt_lang_ptr->second;
-
-
-        // Reset the ggml_context
-        model.ctx = ctx_from_buffer(encoder_buf);
-        ggml_set_no_alloc(model.ctx, true);
+        // Load audio input
         GGML_ASSERT(info.samplerate == 16000);
         GGML_ASSERT(info.channels == 1);
-        // Truncate audio input. Ideally we should chunk it, but this will prevent most obvious OOM.
-        int n_frames = std::min(info.samplerate * params.max_audio_s, (int)info.frames);
-        ggml_tensor* seqs = ggml_new_tensor_2d(model.ctx, GGML_TYPE_F32, n_frames, info.channels);
-        ggml_allocr_alloc(fwd_alloc, seqs);
+        // stop at 30s. Ideally we should chunk input audio, but this will prevent most obvious OOM.
+        int n_frames = std::min(info.samplerate * 30, (int)info.frames);
```

Comment:
max_audio_s was used here to allow longer samples. Agree this is a hack, waiting for a cleaner solution, but we don't have fully decided how to implement this.